### PR TITLE
PLANNER-2822 Correct Quarkus checks and add mandrel LTS check

### PR DESCRIPTION
**Pull request by radtriste**
- setup unique specific nightly jenkinsfile to avoid repetition
- each repository is tested separately
- added Mandrel LTS check

### JIRA

https://issues.redhat.com/browse/PLANNER-2822

### Referenced pull requests

- https://github.com/kiegroup/optaplanner/pull/2217
- https://github.com/kiegroup/optaweb-vehicle-routing/pull/759
- https://github.com/kiegroup/optaplanner-quickstarts/pull/413
- https://github.com/kiegroup/kogito-pipelines/pull/641

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Release notes updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: 
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: 
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] mandrel</b>
</details>

### CI Status

 You can check OptaPlanner repositories CI status from [Chain Status webpage](https://kiegroup.github.io/optaplanner/).

### CODE REVIEW

**Comment by radtriste** [comment]
jenkins run quarkus-branch

**Comment by radtriste** [comment]
test jobs can be found in https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/tradisso/job/optaplanner_nightlies/job/planner-2822/

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -78,21 +78,38 @@ How to retest this PR or trigger a specific build:
...
 - for <b>mandrel checks</b>  
   Run native checks against Mandrel image
   Please add comment: <b>Jenkins run mandrel</b>
 
 - for a <b>specific mandrel check</b>  
   Run native checks against Mandrel image  
-  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts|optaweb-vehicle-routing] mandrel</b>
+  Please add comment: <b>Jenkins (re)run [optaplanner|optaplanner-quickstarts] mandrel</b>
+
+- for <b>mandrel lts checks</b>
```
Comment (rsynek) [inline .github/pull_request_template.md:105]: I assume this runs the check for both the optaplanner and quickstarts?

File: .ci/jenkins/dsl/jobs.groovy [hunk 2]
```diff
@@ -258,6 +213,23 @@ KogitoJobUtils.createVersionUpdateToolsJob(this, 'optaplanner', 'Drools', [
   properties: [ 'version.org.drools' ],
 ])
 
+void setupSpecificNightlyJob(Folder specificNightlyFolder) {
```
Comment (rsynek) [inline .ci/jenkins/dsl/jobs.groovy:216]: If I get it right, while the existing `Jenkinsfile.nightly` is a high-level pipeline that triggers the `Jenkinsfile.deploy`, this new "specific" pipelines just test the single repository with native, Quarkus, and Mandrel.

**Review by rsynek — APPROVED** [review]
I see nothing wrong with this PR; on the other hand, I fail to see the code change that added the Mandrel LTS check. Would you point it out to me, please?

File: .ci/jenkins/Jenkinsfile.specific_nightly [hunk 3]
```diff
@@ -93,44 +104,68 @@ void checkoutOptaplannerRepo() {
...
+    return mvnCmd
+}
+
+MavenCommand getFullMavenCommand(String directory) {
+    def mvnCmd = getEcosystemMavenCommand(directory)
+    if (isNative()) {
+        mvnCmd.withProfiles(['native'])
                 .withProperty('quarkus.native.container-build', true)
                 .withProperty('quarkus.native.container-runtime', 'docker')
                 .withProperty('quarkus.profile', 'native') // Added due to https://github.com/quarkusio/quarkus/issues/13341
```
[the remark this answers is not in the document]
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.specific_nightly:135]: Created https://issues.redhat.com/browse/PLANNER-2828 for that

File: .github/pull_request_template.md (hunk 1 above)
Comment (radtriste) [reply .github/pull_request_template.md:105]: yes

File: .ci/jenkins/dsl/jobs.groovy (hunk 2 above)
Comment (radtriste) [reply .ci/jenkins/dsl/jobs.groovy:216]: yep. This `Jenkinsfile.deploy` has much more options that we really don't need onto the specific nightlies.
In the end, I would like to move this all to the build-chain at some point as we are repeating a lot of things between nightlies and PRs...

**Pull request merged**